src/collection.rs

```diff
@@ -543,3 +800,18 @@ impl<'de> Deserialize<'de> for CollectionType {
         }
     }
 }
+
+/// Create header name and header value from read_options
+fn make_header_from_options(
+    document_read_options: Option<DocumentReadOptions>,
+) -> Option<(String, String)> {
```

Comment:
What about using `Option<http::header::HeaderName, http::header::HeaderValue>`?

Or simply use `http::HeaderMap`, and use [`extend()`](https://doc.rust-lang.org/stable/core/iter/trait.Extend.html#tymethod.extend) to append all values to another `HeaderMap`.